Developing-new-features-with-feature-gating.md

```diff
@@ -93,9 +132,37 @@ We use the following principles to determine the value of a feature flags:
 
 - Each feature flag has multiple *rules*, its value changes to the value specified in the first rule that is matched. If no rule is matched, its value remains default.
 - Each rule has multiple *filters*, and a rule is matched only when all of its filters are matched.
-- Each filter describes a scenario, there are multiple types of filters like `server mode`, `client type`, so that you can you can describe scenarios like `server mode = dev OR server mode = test`.
-- Each filter can have multiple conditions, and a filter is matched if any of the conditions are matched.
+- Each filter describes a scenario. There are multiple types of filters (like `server mode` and `client type`) which can be used in combination with each other to describe various scenarios (like `server mode = dev OR server mode = test`).
+- Each filter can have multiple conditions, and a filter is matched if any of the conditions specified in the rule are matched.
+
+For example, if we want a feature flag to be enabled only in dev environments, we can configure the flag's rules/filters as shown below.
+
+![Example feature setting](https://i.imgur.com/GnFQ7El.jpg)
+
+## Why do we need feature flags?
+
+Feature flags vastly improve/simplify the development process by allowing developers to safely deploy new features to production. They also allow us to quickly disable features that are causing problems in production.
+
+They allow us to hide features that are still in development/still being tested. This enables a developer to add a feature incrementally to the codebase, and only enable it when it is ready for production. Should the feature break, they also allow us to easily disable it without having to roll back all the changes. This is especially useful when the feature is large and/or complex.
+
+## How do you, as a developer, use feature flags?
+
+### When to use feature flags?
+
+Feature flags should be used when you are working on a feature whose scale makes it hard to implement it all at once and will need to be added incrementally/if the feature is likely to cause breakages. Essentially, feature flags are a must for featues that are not ready for production/fully-tested yet.
```

Comment:
...whose scale makes it hard to implement all at once, or for more experimental features that are likely to cause breakages. Essentially, ...